Source/WebCore/dom/TrustedTypePolicyFactory.cpp

```diff
@@ -100,21 +101,59 @@ Ref<TrustedScript> TrustedTypePolicyFactory::emptyScript() const
     return TrustedScript::create(""_s);
 }
 
-String TrustedTypePolicyFactory::getAttributeType(const String& tagName, const String& attribute, const String& elementNamespace, const String& attrNamespace) const
+String TrustedTypePolicyFactory::getAttributeType(const String& tagName, const String& attributeParameter, const String& elementNamespace, const String& attributeNamespace) const
 {
-    UNUSED_PARAM(tagName);
-    UNUSED_PARAM(attribute);
-    UNUSED_PARAM(elementNamespace);
-    UNUSED_PARAM(attrNamespace);
-    return String();
+    auto localName = tagName.convertToASCIILowercase();
+    auto attributeName = attributeParameter.convertToASCIILowercase();
+
+    if (attributeName.startsWith("on"_s))
+        return trustedTypeToString(TrustedType::TrustedScript);
+
+    AtomString elementNS = elementNamespace.isEmpty() ? HTMLNames::xhtmlNamespaceURI : AtomString(elementNamespace);
+    AtomString attributeNS = attributeNamespace.isEmpty() ? nullAtom() : AtomString(attributeNamespace);
+
+    QualifiedName element(nullAtom(), AtomString(localName), elementNS);
+    QualifiedName attribute(nullAtom(), AtomString(attributeName), attributeNS);
+
+    if (element.matches(HTMLNames::embedTag) && attribute.matches(HTMLNames::srcAttr))
+        return trustedTypeToString(TrustedType::TrustedScriptURL);
+    if (element.matches(HTMLNames::iframeTag) && attribute.matches(HTMLNames::srcdocAttr))
+        return trustedTypeToString(TrustedType::TrustedHTML);
+    if (element.matches(HTMLNames::objectTag) && (attribute.matches(HTMLNames::codebaseAttr) || attribute.matches(HTMLNames::dataAttr)))
+        return trustedTypeToString(TrustedType::TrustedScriptURL);
+    if (element.matches(HTMLNames::scriptTag) && (attribute.matches(HTMLNames::srcAttr) || attribute.matches(HTMLNames::dataAttr)))
+        return trustedTypeToString(TrustedType::TrustedScriptURL);
+    if (element.matches(SVGNames::scriptTag) && (attribute.matches(SVGNames::hrefAttr) || attribute.matches(XLinkNames::hrefAttr)))
+        return trustedTypeToString(TrustedType::TrustedScriptURL);
+
+    return nullString();
 }
 
-String TrustedTypePolicyFactory::getPropertyType(const String& tagName, const String& attribute, const String& elementNamespace) const
+String TrustedTypePolicyFactory::getPropertyType(const String& tagName, const String& property, const String& elementNamespace) const
```

Comment:
Looks like a spec has a typo?
https://w3c.github.io/trusted-types/dist/spec/#dom-trustedtypepolicyfactory-getpropertytype
should probably say "If interface has an IDL [attribute](https://webidl.spec.whatwg.org/#dfn-attribute) member which identifier is *property*," not "*attribute*",

Reply:
Good spot, I've got a spec PR that rewrites this function I'll check that it's correct on my PR.